src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -313,13 +313,13 @@ public Settings updatedSettings() {
     synchronized public PluginsInfo info() {
         if (refreshInterval.millis() != 0) {
             if (cachedPluginsInfo != null &&
-                    (refreshInterval.millis() < 0 || (System.currentTimeMillis() - lastRefresh) < refreshInterval.millis())) {
+                    (refreshInterval.millis() < 0 || TimeValue.nsecToMSec(System.nanoTime() - lastRefreshNS) < refreshInterval.millis())) {
```

Comment:
Why not `(System.nanoTime() - lastRefreshNS) < refreshInterval.nanos())` to avoid the extra call to ns->ms conversion?